### PR TITLE
output-fix-esdt-revert

### PR DESCRIPTION
**Pull request by sasurobert**
gas usage fix for esdt and revert esdt transfer in case of errors

### CODE REVIEW

File: arwen/host/execution.go [hunk 1]
```diff
@@ -626,6 +667,12 @@ func (host *vmHost) callBuiltinFunction(input *vmcommon.ContractCallInput) (*vmc
 		return nil, err
 	}
 
+	if newVMInput != nil {
```
Comment (SebastianMarian) [inline arwen/host/execution.go:670]: Coud be err and newVMInput both nil ?

Comment (sasurobert) [reply arwen/host/execution.go:670]: yes. it is possible.

Comment (sasurobert) [reply arwen/host/execution.go:670]: yes. it is

**Review by SebastianMarian — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

File: arwen/host/execution.go [hunk 2]
```diff
@@ -596,6 +601,42 @@ func (host *vmHost) callSCMethodIndirect() error {
...
+			Arguments:      input.Arguments[:2],
+			CallValue:      big.NewInt(0),
+			CallType:       vmcommon.DirectCall,
+			GasPrice:       input.GasPrice,
+			GasProvided:    input.GasProvided,
+			GasLocked:      0,
+			OriginalTxHash: input.OriginalTxHash,
+			CurrentTxHash:  input.CurrentTxHash,
+			ESDTValue:      big.NewInt(0),
+			ESDTTokenName:  nil,
```
Comment (dorin-iancu) [inline arwen/host/execution.go:624]: If this is meant to revert an ESDT transfer, shouldn't this have the original call's ESDTValue and ESDTToken name?

**Review by dorin-iancu — APPROVED** [review]

File: arwen/host/execution.go (hunk 2 above)
Comment (sasurobert) [reply arwen/host/execution.go:624]: it does not matter. built in function does not get it from esdtvalue / esdttokenname but from function and arguments.

**Pull request merged**